packages/core/http/core-http-server/src/versioning/types.ts

```diff
@@ -228,7 +228,7 @@ export interface AddVersionOpts<P, Q, B> {
    * Validation for this version of a route
    * @public
    */
-  validate: false | FullValidationConfig<P, Q, B>;
+  validate: false | FullValidationConfig<P, Q, B> | (() => FullValidationConfig<P, Q, B>); // Provide a way to lazily load validation schemas
```

Comment:
This will enable ResponseOps and other plugins to provide route schemas lazily. Useful for registries that only know schema shapes after setup/start time.

Reply:
I'm surprised to not see associated code changes to support passing a function now. Unless I missed it or it already worked and just wasn't documented (but traditional routes don't support that AFAIK, so...)

Reply:
We have some new code in `packages/core/http/core-http-router-server-internal/src/versioned_router/core_versioned_route.ts` to extract the validations.

But yeah, we should probably make IRouter symmetrical! Will update and add a couple of tests.

Reply:
Pushed some updates that contain tests and update `Router` type and implementatio to support lazy validation